providers/flagd/pkg/service/in_process/service.go

```diff
@@ -334,19 +338,31 @@ func makeSyncProvider(cfg Configuration, log *logger.Logger) (sync.ISync, string
 func mapError(flagKey string, err error) of.ResolutionError {
 	switch err.Error() {
 	case model.FlagNotFoundErrorCode:
-		return of.NewFlagNotFoundResolutionError(fmt.Sprintf("flag: " + flagKey + " not found"))
+		return of.NewFlagNotFoundResolutionError(fmt.Sprintf("flag: %s not found", flagKey))
 	case model.FlagDisabledErrorCode:
-		return of.NewFlagNotFoundResolutionError(fmt.Sprintf("flag: " + flagKey + " is disabled"))
+		return of.NewFlagNotFoundResolutionError(fmt.Sprintf("flag: %s is disabled", flagKey))
 	case model.TypeMismatchErrorCode:
-		return of.NewTypeMismatchResolutionError(fmt.Sprintf("flag: " + flagKey + " evaluated type not valid"))
+		return of.NewTypeMismatchResolutionError(fmt.Sprintf("flag: %s evaluated type not valid", flagKey))
 	case model.ParseErrorCode:
-		return of.NewParseErrorResolutionError(fmt.Sprintf("flag: " + flagKey + " parsing error"))
+		return of.NewParseErrorResolutionError(fmt.Sprintf("flag: %s parsing error", flagKey))
 	default:
-		return of.NewGeneralResolutionError(fmt.Sprintf("flag: " + flagKey + " unable to evaluate"))
+		return of.NewGeneralResolutionError(fmt.Sprintf("flag: %s unable to evaluate", flagKey))
 	}
 }
 
 func isValidTargetScheme(targetUri string) bool {
 	regx := regexp.MustCompile("^" + grpc.SupportedScheme)
 	return regx.Match([]byte(targetUri))
 }
+
+// mergeContexts combines context values with provided context
+func mergeContexts(evalCtx, contextValues map[string]any) map[string]any {
+	merged := make(map[string]any)
+	for k, v := range evalCtx {
```

Comment:
i am not sure about the order here, what has higher importance, i need to check this